plugins/s3fs_check_success.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+import boto3
+from airflow.exceptions import AirflowException
+from airflow.operators import BaseSensorOperator
+from airflow.plugins_manager import AirflowPlugin
+
+
+def check_s3fs_success(bucket, prefix, num_partitions):
+    """Check the s3 filesystem for the existence of `_SUCCESS` files in dataset partitions.
+    
+    :prefix:            Bucket prefix of the table
+    :num_partitions:    Number of expected partitions
+    """
+    s3 = boto3.resource("s3")
+    objects = s3.Bucket(bucket).objects.filter(Prefix=prefix)
```

Comment:
Looking around, I think boto3 implements exponential retries [by default](https://github.com/boto/botocore/blob/2fb16151f80259190f3dc349e01143ea5c480ef3/botocore/data/_retry.json#L83-L104). 

I'll file a follow-up on the default pagination size, it would cause issues for a data-set with a large number of partitions. For example, this is might fail in the case of `main_summary_all_histograms` where I found at least one day with more than 1000 objects.

Reply:
FWIW, I definitely saw sensor jobs fail due to rate limits when I tested out sensor tasks during the main summary backfill – I'll have to check if those use the default retries, but in any case, I do think it's pretty likely we'll run into this unless we increase the number of retries to wait out the rate limiting period

Reply:
A potential failure cases that I'm anticipating (if there are any) is an exception causing the sensor job to prematurely fail. That being said, if the sensor continues poking, then it might be fine. I'll be watching the tasks tonight to reset the status and disable the DAG if need-be.